### DISCUSSION TITLE
After configuring the reverse proxy, events are dispatched but no response occurs.

### DISCUSSION THREAD

**Discussion by kjxbyz** [socketio/socket.io#4929] [Q&A]
After configuring the reverse proxy, the `hello2` event is dispatched, but there is no response.

![WX20240123-084642](https://github.com/socketio/socket.io/assets/47768002/e327082b-1227-4606-ae9b-4d375d232960)

```
server {
    listen 80;
    server_name dev.chenyifaer.com;

    return 301 https://dev.chenyifaer.com$request_uri;
}

server {
    listen 443 ssl http2;
    server_name dev.chenyifaer.com;
    charset     utf-8;

    ssl_certificate     /certs/dev.chenyifaer.com.pem;
    ssl_certificate_key /certs/dev.chenyifaer.com.key;
    ssl_session_timeout  5m;
    ssl_verify_client off;
    ssl_protocols SSLv3 TLSv1 TLSv1.1 TLSv1.2;
    ssl_ciphers ECDHE-RSA-AES128-GCM-SHA256:ECDHE:ECDH:AES:HIGH:!NULL:!aNULL:!MD5:!ADH:!RC4:!DH:!DHE;
    ssl_prefer_server_ciphers  on;

    add_header Access-Control-Allow-Origin *;

    location ^~ /homing-pigeon/ws {
        proxy_pass         https://dev.external.homing-pigeon.chenyifaer.com/ws;
        proxy_set_header   Host                 dev.external.homing-pigeon.chenyifaer.com;
        proxy_set_header   X-Real-IP            $remote_addr;
        proxy_set_header   X-Forwarded-For      $proxy_add_x_forwarded_for;
        proxy_redirect off;
        proxy_buffering off;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection 'upgrade';
        proxy_cache_bypass $http_upgrade;
    }

    location ^~ /socket.io/ {
        set $cors_origin "";
        if ($http_origin ~* "^http://127.0.0.1$") {
            set $cors_origin $http_origin;
        }
        if ($http_origin ~* "^http://localhost$") {
            set $cors_origin $http_origin;
        }
        if ($http_origin ~* "^https://dev.chenyifaer.com$") {
            set $cors_origin $http_origin;
        }
        if ($http_origin ~* "^https://www.chenyifaer.com$") {
            set $cors_origin $http_origin;
        }
        add_header Access-Control-Allow-Origin $cors_origin;
        add_header Access-Control-Allow-Methods 'GET, POST, PATCH, PUT, DELETE';
        add_header Access-Control-Allow-Headers '*';
        add_header Access-Control-Allow-Credentials 'true';

        proxy_pass         https://dev.external.homing-pigeon.chenyifaer.com/socket.io/;
        proxy_set_header   Host                 dev.external.homing-pigeon.chenyifaer.com;
        proxy_set_header   X-Real-IP            $remote_addr;
        proxy_set_header   X-Forwarded-For      $proxy_add_x_forwarded_for;
        proxy_set_header   X-Forwarded-Proto    $scheme;
        proxy_set_header   X-Forwarded-Host     dev.chenyifaer.com;
        proxy_set_header   X-Forwarded-Port     $server_port;
        proxy_set_header   X-NginX-Proxy        true;
        proxy_redirect off;

        proxy_buffers 8 32k;
        proxy_buffer_size 64k;

        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection 'upgrade';
        proxy_cache_bypass $http_upgrade;
        proxy_ssl_server_name on;
    }

}
```

**Comment by darrachequesne** [comment]
Hi! Could you please explain what you are trying to achieve? Why do you have two distinct locations in your nginx configuration? Is `/homing-pigeon/ws` pointing towards your Socket.IO server?

References:

- https://socket.io/docs/v4/reverse-proxy/#nginx
- https://socket.io/docs/v4/troubleshooting-connection-issues/

**Reply by kjxbyz** [reply]
`www.chenyifaer.com` is the main website, and `homing-pigeon` is one of its products. I think the back-end interfaces are all under the `/homing-pigeon` path, such as `/homing-pigeon/api` or `/homing-pigeon/ws`.

`/homing-pigeon/ws` has pointed to the socket server, and the connection can be successful using postman

**Reply by darrachequesne** [reply]
Could you please check if you are able to reach the Socket.IO server with `curl "https://dev.chenyifaer.com/homing-pigeon/ws/?EIO=4&transport=polling"`?

Edit: added the quotes

**Reply by kjxbyz** [reply]
EDIT:
<img width="785" alt="2024-01-24 02 37 48" src="https://github.com/socketio/socket.io/assets/47768002/775ea0a9-5a2d-4b63-bb02-54f44052a7cc">

**Reply by kjxbyz** [reply]
This should be the case in the development environment.

<img width="1108" alt="2024-01-23 23 51 50" src="https://github.com/socketio/socket.io/assets/47768002/6915fe14-2bdf-4d45-8231-b6248cfcd60d">

**Reply by kjxbyz** [reply]
I only configured `websocket` for the `transports` attribute.

![image](https://github.com/socketio/socket.io/assets/47768002/0ac9ce16-dd79-4984-8cd5-c31b03ca32a9)

**Reply by kjxbyz** [reply]
local development environment:


<img width="846" alt="2024-01-24 19 02 22" src="https://github.com/socketio/socket.io/assets/47768002/1d7c12c4-2938-4ecd-9842-3e29ede0841f">

When accessing the following two URLs, the results returned are actually different. Do not you know why?

`curl "http://127.0.0.1:3001/ws/?EIO=4&transport=polling"`

`curl "http://localhost:3001/ws/?EIO=4&transport=polling"`

<img width="1033" alt="2024-01-24 19 08 27" src="https://github.com/socketio/socket.io/assets/47768002/71c8e0d6-b432-42b1-a664-842ac04d0440">

**Reply by darrachequesne** [reply]
I think you are mixing the "namespace" and the "path" concepts here.

The "path" indicates the path of the HTTP request, by default it's `/socket.io/`.

If you want to catch requests on `http://.../ws/`, you will need to set the path to `/ws/`.

Reference: https://socket.io/docs/v4/server-options/#path

The "namespace", on the other hand, is the name of the Socket.IO namespace, which is not found in the request path, but in the Socket.IO packets.

Reference: https://socket.io/docs/v4/namespaces/

**Reply by kjxbyz** [reply]
If you remove the `namespace`, use the default configuration, and set `path` to `/ws/`, then an error will be reported when accessing `http://127.0.0.1:3001/ws/` in postman. You must access `http://127.0.0.1:3001` and set `Handshake path` to `/ws` for normal access.

<img width="1111" alt="2024-01-24 22 49 08" src="https://github.com/socketio/socket.io/assets/47768002/3604e4d6-2121-4a0b-b73b-003ffd5fdb03">

<img width="1151" alt="2024-01-24 22 50 59" src="https://github.com/socketio/socket.io/assets/47768002/7058c388-7cb5-4b0b-a18e-b62dd0a12036">

![WX20240124-225346@2x](https://github.com/socketio/socket.io/assets/47768002/39370f88-752a-4def-b0c0-63c3872a9b71)

**Reply by kjxbyz** [reply]
What makes people confused is that when the socket event is triggered in `postman`, there is no response. However, I used the `socket_io_client` library in `flutter` and found that the response can be received normally.

![WX20240131-225809@2x](https://github.com/socketio/socket.io/assets/47768002/c8b7dd09-018c-4d51-9925-7f8b90494b33)

![WX20240131-225852@2x](https://github.com/socketio/socket.io/assets/47768002/947c3daa-b93a-4c11-ab59-5452ddf22dff)

**Reply by darrachequesne** [reply]
@kjxbyz Thanks for the update :+1: we should definitely add a how-to guide with Postman.
